.github/workflows/weekly-tests.yaml

```diff
@@ -77,3 +77,34 @@ jobs:
         path: output.zip
         retention-days: 7
     - run: echo "This job's status is ${{ job.status }}."
+
+  dramsys-tests:
+    runs-on: [self-hosted, linux, x64, build]
+    container: gcr.io/gem5-test/ubuntu-22.04_all-dependencies:latest
+    timeout-minutes: 4320 # 3 days
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Scheduled workflows run on the default branch by default. We
+          # therefore need to explicitly checkout the develop branch.
+          ref: develop
+
+      - name: Checkout DRAMSys
+        working-directory: ${{ github.workspace }}/ext/dramsys
+        run: |
+          git clone https://github.com/tukl-msd/DRAMSys DRAMSys
+          cd DRAMSys
+          git checkout -b gem5 09f6dcbb91351e6ee7cadfc7bc8b29d97625db8f
+          git submodule update --init --recursive
+
+      # gem5 is built separately because it depends on the DRAMSys library
+      - name: Build gem5
+        working-directory: ${{ github.workspace }}
+        run: scons build/ALL/gem5.opt -j4
```

Comment:
I'm not sure how we do this elsewhere in out workflow files but shouldn't we use something like `nproc` here?

Reply:
Ahh, yeah we do, I can update that! I think it was something I forgot to change when I was testing